vector/src/main/java/im/vector/app/features/home/HomeActivity.kt

```diff
@@ -269,6 +272,48 @@ class HomeActivity :
         navigator.openAnalyticsOptIn(this)
     }
 
+    /**
+     * Migrating from old threads io.element.thread to new m.thread needs an initial sync to
+     * sync and display existing messages appropriately
+     */
+    private fun migrateThreadsIfNeeded(checkSession: Boolean) {
+        if (checkSession) {
+            // We should check session to ensure we will only clear cache if needed
+            val args = intent.getParcelableExtra<HomeActivityArgs>(Mavericks.KEY_ARG)
+            if (args?.existingSession == true) {
+                // existingSession --> Will be true only if we came from an existing active session
+                Timber.i("----> Migrating threads from an existing session..")
+                handleThreadsMigration()
+            } else {
+                // We came from a new session and not an existing one,
+                // so there is no need to migrate threads while an initial synced performed
+                Timber.i("----> No thread migration needed, we are ok")
+                vectorPreferences.threadsMigrated()
+            }
+        } else {
+            // Proceed with migration
+            handleThreadsMigration()
+        }
+    }
+
+    /**
+     * Clear cache and restart to invoke an initial sync for threads migration
+     */
+    private fun handleThreadsMigration() {
+        Timber.i("----> Threads Migration detected, clearing cache and sync...")
+        vectorPreferences.threadsMigrated()
+        MainActivity.restartApp(this, MainActivityArgs(clearCache = true))
+    }
+
+    private fun handleNotifyUserForThreadsMigration() {
+        MaterialAlertDialogBuilder(this)
+                .setTitle(R.string.threads_notice_migration_title)
+                .setMessage(R.string.threads_notice_migration_message)
+                .setCancelable(true)
+                .setPositiveButton(R.string.sas_got_it) { _, _ -> }
```

Comment:
should/does something happen after pressing `got it`?

Reply:
Nothing, it is just informational

Reply:
ah, so the next time they launch the app they'll fall in to the other branch which does the migration?

not a blocker for the PR but I wonder if the dialog could be "migrate now or maybe later"

Reply:
It will not let the user decide to migrate, the notice is just informative. Users that had threads enable will receiver this notice, that will inform them that old messages will be visible in the main timeline and will be automatically migrated to live threads.

Reply:
this is the bit that confuses me, when does the automatic migration happen for these users?

Reply:
At once they open the updated app.
- I am a user and I have lab threads enabled
- I update the app and the migration will happen on the first app run

Is it more clear now?

Reply:
:+1: yep thanks! the message is telling users they have been migrated